ui/src/app/views/notebook-redirect/component.ts

```diff
@@ -175,8 +179,19 @@ export class NotebookRedirectComponent implements OnInit, OnDestroy {
         return localizeObs.retry(3);
       })
       .subscribe((nbName) => {
-        this.progress = Progress.Redirecting;
-        this.window.location.href = this.notebookUrl(this.cluster, nbName);
+        this.incrementProgress(Progress.Redirecting);
+        if (this.creating) {
+          window.history.replaceState({}, '', 'workspaces/' + this.wsNamespace +
+          '/' + this.wsId + '/notebooks/' + this.fullNotebookName);
```

Comment:
encodeUriComponent

Reply:
So I think the replaceState function handles that (since it has in all manual and automated testing). But will add to triple make sure.

Reply:
I suspect this does automatically encode the URI if required just as setting the window.location also does that. This is not sufficient to cover the scenario in my test case in the test case I've been commenting on. If that comment thread still doesn't make sense, we can discuss more on slack.